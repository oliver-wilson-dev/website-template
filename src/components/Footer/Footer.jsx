import React from 'react';
import styles from './Footer.module.css';

const Footer = () => (
  <footer className={styles.footer}>
    <div className={styles.footerContainer}>
      <div className={styles.textContainer}>
        <span className={styles.footerDisclaimerText}>
          This site doesn&apos;t use cookies... yet.
        </span>
        <div className={styles.footerDisclaimerText}>
          Icons made by
          {' '}
          <a className={styles.link} href="https://www.flaticon.com/authors/freepik" title="Freepik">Freepik</a>
          {' '}
          from
          {' '}
          <a className={styles.link} href="https://www.flaticon.com/" title="Flaticon">www.flaticon.com</a>
        </div>
        <span className={styles.footerDisclaimerText}>
          The code for this website can be found on
          {' '}
          <a className={styles.link} href="https://github.com/oliver-wilson-dev/website-template" target="_blank" rel="noopener noreferrer">github</a>
          .
        </span>
        <span className={styles.footerDisclaimerText}>Â©2021, Oliver Wilson</span>
      </div>
    </div>
  </footer>
);

export default Footer;
