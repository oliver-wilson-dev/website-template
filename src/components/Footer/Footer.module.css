@import "../../styles/globalVars.css";
@import "../../styles/colours.css";

:root {
    --fontSize: 12px;
  }

.footer {
    width: 100%;
    display: flex;
    padding: 20px;
    box-sizing: border-box;
    background-color: $dark-grey;
    justify-content: center;
}

.textContainer {
    display: flex;
    flex-basis: 50%;
    flex-direction: column;
    align-items: flex-start;
    justify-content: flex-start;
}

.footerContainer {
    width: 100%;
    max-width: $max-content-width;
}

.link {
    color: #fcbf49;
}

.footerDisclaimerText {
    text-align: left;
    margin-bottom: 20px;
    font-size: var(--fontSize);
    color: white;
  
    &:last-child {
      margin-bottom: 0px;
    }
}