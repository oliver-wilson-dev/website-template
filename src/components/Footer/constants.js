export const GITHUB_PROFILE_URL = 'https://github.com/oliver-wilson-dev/';
export const PHONE_NUMBER = '07580081310';
export const EMAIL_ADDRESS = 'contact@oliverwilson.dev';
