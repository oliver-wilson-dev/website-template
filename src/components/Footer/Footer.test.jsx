import React from 'react';
import { shallow } from 'enzyme';
import Footer from './Footer';

const render = () => shallow(<Footer />);

it('renders correctly', () => {
  expect(render()).toMatchSnapshot();
});
