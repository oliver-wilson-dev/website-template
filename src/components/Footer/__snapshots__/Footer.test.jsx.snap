// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<footer
  className="footer"
>
  <div
    className="footerContainer"
  >
    <div
      className="textContainer"
    >
      <span
        className="footerDisclaimerText"
      >
        This site doesn't use cookies... yet.
      </span>
      <div
        className="footerDisclaimerText"
      >
        Icons made by
         
        <a
          className="link"
          href="https://www.flaticon.com/authors/freepik"
          title="Freepik"
        >
          Freepik
        </a>
         
        from
         
        <a
          className="link"
          href="https://www.flaticon.com/"
          title="Flaticon"
        >
          www.flaticon.com
        </a>
      </div>
      <span
        className="footerDisclaimerText"
      >
        The code for this website can be found on
         
        <a
          className="link"
          href="https://github.com/oliver-wilson-dev/website-template"
          rel="noopener noreferrer"
          target="_blank"
        >
          github
        </a>
        .
      </span>
      <span
        className="footerDisclaimerText"
      >
        Â©2021, Oliver Wilson
      </span>
    </div>
  </div>
</footer>
`;
